### PR TITLE
[do not merge] Sync AK/trunk to CCS/master Dec 2022

### PR DESCRIPTION
**Pull request by gbadoni**
NOTE: This PR should be merged using CLI to preserve commit history

-remote check
❯ git remote -v
ak	git@github.com:apache/kafka.git (fetch)
ak	git@github.com:apache/kafka.git (push)
origin	git@github.com:confluentinc/kafka.git (fetch)
origin	git@github.com:confluentinc/kafka.git (push)

- create branch for merge
git pull origin master
git checkout -b sync-ak-to-ccs-Dec-2022

- merge conflicts
git merge ak/trunk

- fix conflicts
git status
On branch sync-ak-to-ccs-Dec-2022
All conflicts fixed but you are still merging.
Unmerged paths:
  (use "git add <file>..." to mark resolution)
	both modified:   Jenkinsfile
	both modified:   core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala
	both modified:   gradle.properties
	both modified:   tests/kafkatest/__init__.py
	both modified:   tests/kafkatest/version.py

-- Conflict details
- gradle.properties/__init__.py and version.py: Fix the version conflict, pick the version 7.4.0-0-ccs
image

- JenkinsFile : Kept the CCS block of the code. Looks like the AK version wants to send an email to dev@kafka.apache.org, which we do not want. Please review. 
image

- core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala: This is a scala version upgrade change done as a part of the following PR. Keeping the AK code block and tested that build passed.
https://github.com/apache/kafka/pull/12675/files#diff-73a6c1d4cb03750d4a7dca9fe5e5c012b8b1949b73a15da3aaea8714a54cfc5f. 

image

### CODE REVIEW

**Comment by jolshan** [comment]
fyi this did still send an email to the AK dev mailing list -- 😅 hopefully we fixed it to not do that in the future.

File: core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala [hunk 1]
```diff
@@ -201,9 +201,9 @@ class TopicCommandTest {
       () => topicService.deleteTopic(new TopicCommandOptions(Array("--topic", topicName))))
     assertTrue(exception.getCause.isInstanceOf[ThrottlingQuotaExceededException])
 
-    verify(adminClient, times(1)).deleteTopics(
-      argThat((_.contains(topicName)): ArgumentMatcher[Collection[String]]),
-      argThat((_.shouldRetryOnQuotaViolation() == false): ArgumentMatcher[DeleteTopicsOptions])
+    verify(adminClient).deleteTopics(
+      argThat((topics: java.util.Collection[String]) => topics.asScala.toBuffer.equals(Seq(topicName))),
```
Comment (jolshan) [inline core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala:205]: Do we know why ccs had "contains" rather than equals?

**Comment by ijuma** [comment]
The automatic merge was unblocked by https://github.com/confluentinc/kafka/pull/850, so we don't need this anymore.

File: core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala (hunk 1 above)
Comment (ijuma) [reply core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala:205]: The upstream change is the right one, a slightly different change had been applied to ccs which made the test assertion unnecessarily weaker.

**Pull request closed without merging**